File: sapl-server-ce/frontend/create-variable.js
/*
 * Copyright (C) 2017-2023 Dominic Heutelbeck (dominic@heutelbeck.com)
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-vertical-layout.js';
import '@vaadin/vaadin-ordered-layout/src/vaadin-horizontal-layout.js';
import '@vaadin/vaadin-button/src/vaadin-button.js';
import '@vaadin/vaadin-text-field/src/vaadin-text-field.js';

class CreateVariable extends PolymerElement {

    static get template() {
        return html`
<style include="shared-styles">
                :host {
                    display: block;
                    height: 100%;
                }
            </style>
<vaadin-vertical-layout style="width: 100%; height: 100%; padding: var(--lumo-space-m);" theme="spacing">
 <h1 style="margin: var(--lumo-space-s); padding: var(--lumo-space-s);">Create Variable</h1>
 <vaadin-text-field id="nameTextField" style="flex-shrink: 1; margin: var(--lumo-space-s); padding: var(--lumo-space-s); align-self: stretch;" minlength="1" maxlength="50" label="Name"></vaadin-text-field>
 <vaadin-horizontal-layout style="align-self: flex-end; flex-grow: 0;" theme="spacing">
  <vaadin-button id="createButton" theme="primary">
   Create
  </vaadin-button>
  <vaadin-button id="cancelButton">
   Cancel
  </vaadin-button>
 </vaadin-horizontal-layout>
</vaadin-vertical-layout>
`;
    }

    static get is() {
        return 'create-variable';
    }

    static get properties() {
        return {
            // Declare your properties here.
        };
    }
}

customElements.define(CreateVariable.is, CreateVariable);

File: sapl-server-ce/src/main/java/io/sapl/server/ce/ui/views/pdpconfig/CreateVariable.java
/*
 * Copyright (C) 2017-2025 Dominic Heutelbeck (dominic@heutelbeck.com)
 *
 * SPDX-License-Identifier: Apache-2.0
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.sapl.server.ce.ui.views.pdpconfig;

import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.component.textfield.TextField;
import io.sapl.api.SaplVersion;
import lombok.Setter;

import java.io.Serial;

public class CreateVariable extends VerticalLayout {

    @Serial
    private static final long serialVersionUID = SaplVersion.VERSION_UID;

    private final TextField nameTextField = new TextField("Variable Name");
    private final Button    createButton  = new Button("Create");
    private final Button    cancelButton  = new Button("Cancel");

    @Setter
    private transient UserConfirmedListener userConfirmedListener;

    public CreateVariable() {
        initUi();
    }

    public String getNameOfVariableToCreate() {
        return nameTextField.getValue();
    }

    private void initUi() {
        nameTextField.setPlaceholder("name");
        createButton.addClickListener(e -> setConfirmationResult(true));
        cancelButton.addClickListener(e -> setConfirmationResult(false));
        nameTextField.focus();

        final var buttonLayout = new HorizontalLayout(cancelButton, createButton);
        add(nameTextField, buttonLayout);
    }

    private void setConfirmationResult(boolean isConfirmed) {
        if (userConfirmedListener != null) {
            userConfirmedListener.onConfirmationSet(isConfirmed);
        }
    }

    public interface UserConfirmedListener {
        void onConfirmationSet(boolean isConfirmed);
    }
}
